### PR TITLE
STRATCONN-4169 | Events to Klaviyo destination is failing for delivery

### PR DESCRIPTION
**Pull request by Innovative-GauravKochar**
This Pull request is to fix the event delivery issue in TrackEvent and OrderCompleted Actions of Klaviyo (Actions)due to time property. So we are converting the timestamp to ISO Format .To understand more about the problem and solution , Please do go through this testing doc [here](https://docs.google.com/document/d/1lKKg4pyDrTf52N_B89ABO-GSE-YY6_SsWg-vccEANS8/edit#heading=h.x9dgiyfuowsl).

Jira ticket :- https://segment.atlassian.net/jira/software/c/projects/STRATCONN/boards/310?assignee=63617339fc0cc7a600b03c6b&selectedIssue=STRATCONN-4169
## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment . Please have a look [here](https://docs.google.com/document/d/1lKKg4pyDrTf52N_B89ABO-GSE-YY6_SsWg-vccEANS8/edit#heading=h.x9dgiyfuowsl)

### CODE REVIEW

File: packages/destination-actions/src/destinations/klaviyo/functions.ts [hunk 1]
```diff
@@ -408,3 +408,34 @@ export function validatePhoneNumber(phone?: string): boolean {
...
+
+export function roundTimestamp(timestamp: string): string {
+  // Split the timestamp into date and time components
+  const [datePart, timePart] = timestamp.split('T')
+
+  // If there's no timePart, return the original timestamp directly
+  if (!timePart) {
+    return timestamp
+  }
+
```
Comment (itsarijitray) [inline packages/destination-actions/src/destinations/klaviyo/functions.ts:420]: Suggestion: Can you use some date time library like `moment` or `date-fns` for these operations? These would reduce the manual checks and enforce consistency. (optional)

Comment (varadarajan-tw) [reply packages/destination-actions/src/destinations/klaviyo/functions.ts:420]: We have `dayjs` available within actions framework for use already. 

https://github.com/segmentio/action-destinations/blob/c860072c3cd6d45747c09fd29aebdec2c51c2cda/packages/destination-actions/src/lib/dayjs.ts#L6

Comment (itsarijitray) [reply packages/destination-actions/src/destinations/klaviyo/functions.ts:420]: sample code:

```
const dayjs = require('dayjs')
let dayjsDate = dayjs('2024-07-08T18:30:00.89999999Z')

if (!dayjsDate.isValid()) {
  console.log("invalid")
} 
console.log(dayjsDate.toISOString())

```

**Review by itsarijitray — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @Innovative-GauravKochar PR deployed. Please confirm you are happy with the update.

**Comment by Innovative-GauravKochar** [comment]
Thank you @joe-ayoub-segment for the deployment , it's working fine on prod.

**Pull request merged**